src/compiler/checker.ts

```diff
@@ -382,10 +382,15 @@ namespace ts {
 
         /** Returns true if node1 is defined before node 2**/
         function isDefinedBefore(node1: Node, node2: Node): boolean {
+            if (isAliasSymbolDeclaration(node1) && !isInternalModuleImportEqualsDeclaration(node1)) {

```

Comment:
isDefinedBefore is more relaxed kind of implementation so anything that it cant determine is True.